virtwho/virt/ahv/ahv_interface.py

```diff
@@ -557,11 +565,12 @@ def get_vms_uuid(self, version):
                             "Cannot access the uuid for the vm %s. "
                             "vm object: %s" % (vm_entity['name'], vm_entity)
                         )
-
-            body['offset'] = body['offset'] + length
-            body_data = json.dumps(body, indent=4)
-            self._logger.debug('next vm list call has this body: %s' % body)
-            res = self.make_rest_call(method=cmd_method, uri=url, data=body_data)
+            if is_pc is True:
+                body['offset'] = body['offset'] + length
+                self._logger.debug('Next vm list call has this body: %s' % body)
+                res = self.make_rest_call(method=cmd_method, uri=url, json=body)
+            else:
+                res = self.make_rest_call(method=cmd_method, uri=url)
```

Comment:
Note: it is probably not necessary to use this for loop, when version 2.0 is used, because complete list is gathered in first REST API call outside of this for loop. I will try to fix it in different PR.

Reply:
I only hope that Nutanix returns complete list, when version 2.x is used. There is no limitation described in the documentation of API: https://www.nutanix.dev/api_references/prism-v2-0/#/b3A6MjU1ODkwNzg-get-a-list-of-virtual-machines On the other side documentation says that query arguments `offset` and `lengths` could be used.